modules/basegl/include/modules/basegl/processors/raycasting/texturedisosurfacerenderer.h

```diff
@@ -0,0 +1,107 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <modules/basegl/baseglmoduledefine.h>
+
+#include <inviwo/core/datastructures/representationconverter.h>         // for RepresentationCo...
+#include <inviwo/core/datastructures/representationconverterfactory.h>  // for RepresentationCo...
+#include <inviwo/core/processors/processorinfo.h>                       // for ProcessorInfo
+#include <inviwo/core/properties/transferfunctionproperty.h>            // for TransferFunctionPr...
+#include <inviwo/core/util/zip.h>                                       // for zipper
+#include <modules/basegl/processors/raycasting/volumeraycasterbase.h>   // for VolumeRaycasterBase
+#include <modules/basegl/shadercomponents/backgroundcomponent.h>        // for BackgroundComponent
+#include <modules/basegl/shadercomponents/cameracomponent.h>            // for CameraComponent
+#include <modules/basegl/shadercomponents/entryexitcomponent.h>         // for EntryExitComponent
+#include <modules/basegl/shadercomponents/tfcomponent.h>
+#include <modules/basegl/shadercomponents/isocomponent.h>
+#include <modules/basegl/shadercomponents/lightcomponent.h>              // for LightComponent
+#include <modules/basegl/shadercomponents/positionindicatorcomponent.h>  // for PositionIndicato...
+#include <modules/basegl/shadercomponents/raycastingcomponent.h>         // for RaycastingComponent
+#include <modules/basegl/shadercomponents/sampletransformcomponent.h>    // for SampleTransformC...
+#include <modules/basegl/shadercomponents/sphericalcomponent.h>          // for SphericalComponent
+#include <modules/basegl/shadercomponents/volumecomponent.h>             // for VolumeComponent
+
+#include <array>          // for array
+#include <memory>         // for unique_ptr
+#include <string_view>    // for string_view
+#include <unordered_set>  // for unordered_set
+#include <vector>         // for vector
+
+#include <fmt/core.h>  // for format, basic_st...
+
+namespace inviwo {
+
+class IVW_MODULE_BASEGL_API TexturedIsoSurfaceComponent : public ShaderComponent {
+public:
+    TexturedIsoSurfaceComponent(std::string_view volume);
+
+    virtual std::string_view getName() const override;
+    virtual std::vector<std::tuple<Inport*, std::string>> getInports() override;
+    virtual std::vector<Segment> getSegments() override;
+    virtual std::vector<Property*> getProperties() override;
+
+    virtual void initializeResources(Shader& shader) override;
+    virtual void process(Shader& shader, TextureUnitContainer& cont) override;
+
+    std::string volume;
+    VolumeInport colorPort;
+
+    IsoValueProperty* iso;
+    TransferFunctionProperty* tf;
+
+    OptionPropertyInt isoChannel;
+    OptionPropertyInt colorChannel;
+    FloatProperty samplingRate;
+};
+
+class IVW_MODULE_BASEGL_API TexturedIsoSurfaceRenderer : public VolumeRaycasterBase {
```

Comment:
Would be good to describe the functionality of this one. One has to dig quite deep into the code to understand it.